graylog2-web-interface/packages/eslint-plugin-graylog/lib/rules/user-date-time-injection.js

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+/* eslint-disable global-require */
+module.exports = (context) => ({
+  CallExpression: (node) => {
+    const { callee } = node;
+
+    if (callee.name === 'useContext'
+      && callee?.parent?.arguments?.[0].name === 'UserDateTimeContext'
+      && !context.getFilename().endsWith('useUserDateTime.ts')
```

Comment:
Maybe the last condition is too specific and we should just disable the rule in the `useUserDateTime` file instead?